src/test/resources/org/apache/commons/text/names/testNames.txt

```diff
@@ -1,32 +1,33 @@
-Name|LeadingInit|FirstName|NickName|MiddleName|LastName|Suffix
-Björn O'Malley||Björn|||O'Malley|
-Bin Lin||Bin|||Lin|
-Linda Jones||Linda|||Jones|
-Jason H. Priem||Jason||H.|Priem|
-Björn O'Malley-Muñoz||Björn|||O'Malley-Muñoz|
-Björn C. O'Malley||Björn||C.|O'Malley|
-Björn "Bill" O'Malley||Björn|Bill||O'Malley|
-Björn ("Bill") O'Malley||Björn|Bill||O'Malley|
-Björn ("Wild Bill") O'Malley||Björn|Wild Bill||O'Malley|
-Björn (Bill) O'Malley||Björn|Bill||O'Malley|
-Björn 'Bill' O'Malley||Björn|Bill||O'Malley|
-Björn C O'Malley||Björn||C|O'Malley|
-Björn C. R. O'Malley||Björn||C. R.|O'Malley|
-Björn Charles O'Malley||Björn||Charles|O'Malley|
-Björn Charles R. O'Malley||Björn||Charles R.|O'Malley|
-Björn van O'Malley||Björn|||van O'Malley|
-Björn Charles van der O'Malley||Björn||Charles|van der O'Malley|
-Björn Charles O'Malley y Muñoz||Björn||Charles|O'Malley y Muñoz|
-Björn O'Malley, Jr.||Björn|||O'Malley|Jr.
-Björn O'Malley Jr||Björn|||O'Malley|Jr
-B O'Malley||B|||O'Malley|
-William Carlos Williams||William||Carlos|Williams|
-C. Björn Roger O'Malley|C.|Björn||Roger|O'Malley|
-B. C. O'Malley||B.||C.|O'Malley|
-B C O'Malley||B||C|O'Malley|
-B.J. Thomas||B.J.|||Thomas|
-O'Malley, Björn||Björn|||O'Malley|
-O'Malley, Björn Jr||Björn|||O'Malley|Jr
-O'Malley, C. Björn|C.|Björn|||O'Malley|
-O'Malley, C. Björn III|C.|Björn|||O'Malley|III
-O'Malley y Muñoz, C. Björn Roger III|C.|Björn||Roger|O'Malley y Muñoz|III
\ No newline at end of file
+Name|Salutation|LeadingInit|FirstName|NickName|MiddleName|LastName|Suffix
+Björn O'Malley|||Björn|||O'Malley|
+Bin Lin|||Bin|||Lin|
+Linda Jones|||Linda|||Jones|
+Jason H. Priem|||Jason||H.|Priem|
+Björn O'Malley-Muñoz|||Björn|||O'Malley-Muñoz|
+Björn C. O'Malley|||Björn||C.|O'Malley|
+Björn "Bill" O'Malley|||Björn|Bill||O'Malley|
+Björn ("Bill") O'Malley|||Björn|Bill||O'Malley|
+Björn ("Wild Bill") O'Malley|||Björn|Wild Bill||O'Malley|
+Björn (Bill) O'Malley|||Björn|Bill||O'Malley|
+Björn 'Bill' O'Malley|||Björn|Bill||O'Malley|
+Björn C O'Malley|||Björn||C|O'Malley|
+Björn C. R. O'Malley|||Björn||C. R.|O'Malley|
+Björn Charles O'Malley|||Björn||Charles|O'Malley|
+Björn Charles R. O'Malley|||Björn||Charles R.|O'Malley|
+Björn van O'Malley|||Björn|||van O'Malley|
+Björn Charles van der O'Malley|||Björn||Charles|van der O'Malley|
+Björn Charles O'Malley y Muñoz|||Björn||Charles|O'Malley y Muñoz|
+Björn O'Malley, Jr.|||Björn|||O'Malley|Jr.
+Björn O'Malley Jr|||Björn|||O'Malley|Jr
+B O'Malley|||B|||O'Malley|
+William Carlos Williams|||William||Carlos|Williams|
+C. Björn Roger O'Malley||C.|Björn||Roger|O'Malley|
+B. C. O'Malley|||B.||C.|O'Malley|
+B C O'Malley|||B||C|O'Malley|
+B.J. Thomas|||B.J.|||Thomas|
+O'Malley, Björn|||Björn|||O'Malley|
+O'Malley, Björn Jr|||Björn|||O'Malley|Jr
+O'Malley, C. Björn||C.|Björn|||O'Malley|
+O'Malley, C. Björn III||C.|Björn|||O'Malley|III
+O'Malley y Muñoz, C. Björn Roger III||C.|Björn||Roger|O'Malley y Muñoz|III
+Dr. Gaius Baltar|Dr.||Gaius|||Baltar|
```

Comment:
Perfect. Added that test and finally understood your point.

@tmackenzie are you OK if I merge the pull request, then update the code according to @dspasojevic and my comments here? :-)